packages/teuchos/numerics/src/Teuchos_SerialDenseMatrix.hpp

```diff
@@ -82,7 +80,7 @@ class SerialDenseMatrix : public CompObject, public BLAS<OrdinalType, ScalarType
   /*! Creates a empty matrix of no dimension.  The Shaping methods should be used to size this matrix.
     Values of this matrix should be set using the [], (), or = operators.
   */
-  SerialDenseMatrix();
+  SerialDenseMatrix() = default;
```

Comment:
@bartlettroscoe wrote:
> Is a default constructor really safe? There is a raw pointer in this class!

Yes, it's safe, because I added a default member initializer that sets the pointer to `nullptr`.  Constructors -- whether explicitly defined, implicitly defined, or defaulted -- [always respect default member initializers](https://en.cppreference.com/w/cpp/language/data_members#Member_initialization).